File: HoWo7/Phone/Phone.py
from Phone.InterfCall import InterfCall
from Phone.InterfSendSMS import InterfSendSMS


class Phone(InterfCall, InterfSendSMS):
    __model: str
    __sizeDisplay: int  # double
    __akb: int
    __price: int  # double

    def __init__(self, model: str, sizeDisplay, akb: int, price):
        self.__model = model
        self.__sizeDisplay = sizeDisplay
        self.__akb = akb
        self.__price = price
        # super().__init__()

    def getModel(self) -> str:
        return self.__model

    def getSizeDisplay(self):  # -> double :
        return self.__sizeDisplay

    def getAkb(self) -> int:
        return self.__akb

    def getPrice(self):  # -> double:
        return self.__price

    def setPrice(self, price):
        self.__price = price

    def __eq__(self, other) -> bool:
        if not isinstance(other, Phone):
            raise TypeError("Операнд справа должен иметь тип Phone")
        return self.__model == other.getModel() and self.__sizeDisplay == other.getSizeDisplay() and self.__akb == other.getAkb()

    def __str__(self) -> str:
        return f"model={self.__model}, sizeDisplay={self.__sizeDisplay}, akb={self.__akb}, price={self.__price}"

    def charge(self):
        print("Неужели он разрядился")

File: HoWo5_6/Phone/Phone.java
package HoWo5_6.Phone;

public abstract class Phone  implements InterfCarge, InterfCall, InterfSendSMS {

    private String model;
    private double sizeDisplay;
    private int akb;
    private double price;

    public Phone(String model, double sizeDisplay, int akb, double price) {
        this.model = model;
        this.sizeDisplay = sizeDisplay;
        this.akb = akb;
        this.price = price;
    }

    public String getModel() {
        return model;
    }

    public double getSizeDisplay() {
        return sizeDisplay;
    }

    public int getAkb() {
        return akb;
    }

    public double getPrice() {
        return price;
    }

    public void setPrice(double price) {
        this.price = price;
    }

    @Override
    public String toString() {
        return "model=" + this.model + ", sizeDisplay=" + this.sizeDisplay + ", akb=" + this.akb + ", price=" + this.price;
    }

    @Override
    public void charge(){
        //
    }
}
